components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -324,6 +269,102 @@ func (r *NotebookReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	return ctrl.Result{RequeueAfter: culler.GetRequeueTime()}, nil
 }
 
+func updateNotebookStatus(r *NotebookReconciler, nb *v1beta1.Notebook,
+	sts *appsv1.StatefulSet, req ctrl.Request) (bool, error) {
+
+	log := r.Log.WithValues("notebook", req.NamespacedName)
+	ctx := context.Background()
+
+	statusUpdate := false
+	// Initialize Notebook CR Status
+	if nb.Status.Conditions == nil {
+		log.Info("Initializing Notebook CR Status", "namespace", nb.Namespace, "name", nb.Name)
+		newStatus := initializeStatus()
+		nb.Status = newStatus
+		statusUpdate = true
+	}
+
+	// Update the Notebook CR status.readyReplicas if the status is changed
+	if sts.Status.ReadyReplicas != nb.Status.ReadyReplicas {
+		log.Info("Updating status.ReadyReplicas", "namespace", nb.Namespace, "name", nb.Name)
+		nb.Status.ReadyReplicas = sts.Status.ReadyReplicas
+		statusUpdate = true
+	}
+
+	// Check the pod status
+	pod := &corev1.Pod{}
+	podFound := false
+	err := r.Get(ctx, types.NamespacedName{Name: sts.Name + "-0", Namespace: sts.Namespace}, pod)
```

Comment:
Let's move this part completely outside of this function and treat the Pod as the StatefulSet. 

The main reconciliation logic is responsible for finding the objects (Pod, StatefulSet) and our function will need to check if they are `nil` or not